encryption/conn.go

```diff
@@ -0,0 +1,217 @@
+package encryption
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math"
+	"net"
+)
+
+const (
+	// MsgLenFieldSize is the byte size of the frame length field of a
+	// framed message.
+	MsgLenFieldSize = 4
+	// The byte size of the message type field of a framed message.
+	msgTypeFieldSize = 4
+	// The bytes size limit for a ALTS record message.
```

Comment:
I don't think we have to mention `ALTS` in our code, because this means _(Application Layer Transport Security (ALTS) is a Google developed authentication and encryption system)_
* This is our implementation, yes we use their code, but we are using completely different logic.
* Our code is open source, people who is going to invest million dollars, (I believe) they will review thorough the code. Imagine how to surprise them, In other words, our code should look presentable.